packages/block-editor/src/components/link-control/index.js

```diff
@@ -146,22 +152,64 @@ function LinkControl( {
 								onChange( { ...value, ...suggestion } );
 							} }
 							isSelected={ index === selectedSuggestion }
-							isURL={ manualLinkEntryTypes.includes( suggestion.type.toLowerCase() ) }
+							isURL={ directLinkEntryTypes.includes( suggestion.type.toLowerCase() ) }
 							searchTerm={ inputValue }
 						/>
 					) ) }
+
+					{ showCreatePages && createEmptyPage && ! isInitialSuggestions && ! isSingleDirectEntryResult && (
+						<LinkControlSearchCreate
+							searchTerm={ inputValue }
+							onClick={ async () => {
+								setIsResolvingLink( true );
+								const newPage = await createEmptyPage( inputValue );
+								// TODO: handle error from API
+								setIsResolvingLink( false );
+								setIsEditingLink( false );
+								onChange( {
+									id: newPage.id,
+									title: newPage.title.raw, // TODO: use raw or rendered?
+									url: newPage.link,
+									type: newPage.type,
+								} );
```

Comment:
As per other comments should we generalise the terms here away from `page` and make this code more generalised?

Steps to consider:

* Remove `page` references.
* Avoid coupling result of entity create to the concept and schema of a `Page`. Suggest the properties of (what is currently called) `newPage` should be general and already be transformed to match those required by the schema of a `LinKControl` "link" (eg: `id`, `title`, `url`, `type`).

Basically we should require the component consuming `LInkControl` to do more of the heavy lifting around creating and normalising the new entity result and link control should only be concerned with manipulating component state.

Reply:
@marekhrabe It looks like this is now complete right?

Reply:
Haven't specifically hunted for all "page" occurrences but the part with link/entity object shape is done. We can create `posts` right away with this API if we chose to

Reply:
👍 Let's resolve this bit and handle the `Page` references separately.